workflow-runtime/src/main/java/com/squareup/workflow1/RenderingAndSnapshot.kt

```diff
@@ -21,4 +21,21 @@ package com.squareup.workflow1
 data class RenderingAndSnapshot<out RenderingT>(
   val rendering: RenderingT,
   val snapshot: TreeSnapshot
-)
+) {
+
+  override fun equals(other: Any?): Boolean {
```

Comment:
We should write a small test for this equals/hashcode too, now that it's explicit.